Resources/public/js/ezrecommendation.js

```diff
@@ -0,0 +1,77 @@
+// register HandleBars helpers when library is loaded
+
+document.addEventListener('DOMContentLoaded', function() {
+    Handlebars.registerHelper('each', function(context, options) {
+        var result = '';
+        for(var i = 0, j = context.length; i < j; i++) {
+            result = result + options.fn(context[i]);
+        }
+        return result;
+    });
+
+    Handlebars.registerHelper('formatDate', function(timestamp) {
+        var fix = (timestamp + '000') * 1;
+        var t = new Date(fix);
+        return t.toLocaleDateString();
+    });
+
+    Handlebars.registerHelper('formatTime', function(timestamp) {
+        var t = new Date(timestamp);
+        return t.toLocaleTimeString();
+    });
+});
+
+// EzRecoRestClient class
+
+var EzRecoRestClient = function (config) {
+    this.config = config;
+};
+
+EzRecoRestClient.prototype.fetchRecommendations = function (responseCallback, targetId, templateId) {
+    var xmlhttp = new XMLHttpRequest();
+
+    xmlhttp.onreadystatechange = function () {
+        if (xmlhttp.readyState == XMLHttpRequest.DONE) {
+            if (xmlhttp.status == 200) {
+                var jsonResponse = JSON.parse(xmlhttp.response);
```

Comment:
Define variables at the top of the function body. As you can see you initialize the `var jsonResponse` twice, which is incorrect.